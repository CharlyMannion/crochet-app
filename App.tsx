import { Button, StyleSheet, Text, View } from "react-native";

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Crochet!</Text>
      <Text style={styles.dummyText}>Some text about this crochet app!</Text>
      <Button title="tap me"></Button>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  dummyText: {
    margin: 16,
    padding: 16,
    borderColor: "red",
    borderWidth: 1,
  },
});
