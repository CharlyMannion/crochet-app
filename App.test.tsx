import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";

import App from "./App";

describe("<App />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree?.children.length).toBe(3);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders Hello World message on the home page", async () => {
    const screnn = render(<App />);
    expect(screen.getByText("Crochet!")).toBeDefined();
  });
});
