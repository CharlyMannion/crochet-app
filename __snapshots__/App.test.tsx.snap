// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Crochet!
  </Text>
  <Text
    style={
      {
        "borderColor": "red",
        "borderWidth": 1,
        "margin": 16,
        "padding": 16,
      }
    }
  >
    Some text about this crochet app!
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        tap me
      </Text>
    </View>
  </View>
</View>
`;
